api/restHandler/AppListingRestHandler.go

```diff
@@ -561,8 +568,99 @@ func (handler AppListingRestHandlerImpl) RedirectToLinkouts(w http.ResponseWrite
 	}
 	http.Redirect(w, r, link, http.StatusOK)
 }
+func (handler AppListingRestHandlerImpl) fetchResourceTreeFromInstallAppService(w http.ResponseWriter, r *http.Request, appDetail bean.AppDetailContainer) bean.AppDetailContainer {
+	ctx, cancel := context.WithCancel(r.Context())
```

Comment:
take this logic in FetchResourceTree also, since u are passing context, this can be handled there.